src/libAtomVM/externalterm.c

```diff
@@ -390,6 +394,33 @@ static int serialize_term(uint8_t *buf, term t, GlobalContext *glb)
             k += serialize_term(IS_NULL_PTR(buf) ? NULL : buf + k, mfa, glb);
         }
         return k;
+    } else if (term_is_local_pid(t)) {
+        if (!IS_NULL_PTR(buf)) {
+            buf[0] = NEW_PID_EXT;
+        }
+        size_t k = 1;
+        term node_name = glb->node_name;
```

Comment:
let's double check what happens if they are changed on a different thread, maybe they need to be atomic.

Reply:
`glb->node_name` is ATOMIC because of this line. Did I miss something?

Reply:
oh right, my mistake.